sources/packages/backend/apps/queue-consumers/src/services/cas-invoice/cas-invoice.service.ts

```diff
@@ -0,0 +1,264 @@
+        select: {
+          id: true,
+          casSupplier: {
+            id: true,
+            supplierNumber: true,
+            supplierAddress: { supplierSiteCode: true },
+          },
+          createdAt: true,
+          invoiceNumber: true,
+          disbursementReceipt: {
+            id: true,
+            fileDate: true,
+            createdAt: true,
+          },
+          casInvoiceBatch: {
+            id: true,
+            batchName: true,
+            batchDate: true,
+            createdAt: true,
+          },
+          casInvoiceDetails: {
+            id: true,
+            valueAmount: true,
+            casDistributionAccount: {
+              id: true,
+              operationCode: true,
+              awardValueCode: true,
+              distributionAccount: true,
+            },
+          },
+        },
+        relations: {
+          casInvoiceBatch: true,
+          disbursementReceipt: true,
+          casSupplier: true,
+          casInvoiceDetails: {
+            casDistributionAccount: true,
+          },
+        },
+        where: {
+          casInvoiceBatch: {
+            approvalStatus: CASInvoiceBatchApprovalStatus.Approved,
+          },
+          invoiceStatus: CASInvoiceStatus.Pending,
+        },
+      });
+    return pendingInvoices;
+  }
+
+  /**
+   * Send pending invoices to CAS.
+   * @param parentProcessSummary parent process summary for logging.
+   * @param pendingInvoices pending invoices.
+   */
+
+  async sendInvoices(
+    parentProcessSummary: ProcessSummary,
+    pendingInvoices: CASInvoice[],
+  ): Promise<void> {
+    // Process each invoice in parallel.
+    await processInParallel(
+      (pendingInvoice) =>
+        this.processInvoice(pendingInvoice, parentProcessSummary),
+      pendingInvoices,
+    );
+  }
+
+  /**
+   * Process and send an invoice to CAS.
+   * @param pendingInvoice pending invoice.
+   * @param parentProcessSummary parent process summary for logging.
+   */
+  private async processInvoice(
+    pendingInvoice: CASInvoice,
+    parentProcessSummary: ProcessSummary,
+  ): Promise<void> {
+    // Log information about the pending invoice being processed.
+    parentProcessSummary.info(
+      `Processing pending invoice: ${pendingInvoice.invoiceNumber}.`,
+    );
+    const pendingInvoicePayload = this.getPendingInvoicePayload(pendingInvoice);
+    parentProcessSummary.info(
+      `Pending invoice payload invoice number: ${pendingInvoicePayload.invoiceNumber}.`,
+    );
+    let response: SendPendingInvoicesResponse;
+    try {
+      response = await this.casService.sendPendingInvoices(
+        pendingInvoicePayload,
+      );
+      response.invoiceNumber
+        ? parentProcessSummary.info(
+            `Invoice sent to CAS ${response.casReturnedMessages}.`,
+          )
+        : parentProcessSummary.warn(
+            `Invoice ${pendingInvoicePayload.invoiceNumber} send to CAS did not succeed. Reason: ${response.casReturnedMessages}.`,
```

Comment:
If we are getting a HTTP 200, the statement that `send to CAS did not succeed` does not seem accurate.
The message can be adjusted as below.
```
Invoice ${pendingInvoicePayload.invoiceNumber} set to CAS was considered successful but returned additional message(s): ${response.casReturnedMessages}.
```

Reply:
There was a typo in the provided sample. Please review it.